x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/components/add_exception_flyout/index.tsx

```diff
@@ -130,6 +130,16 @@ export const AddExceptionFlyout = memo(function AddExceptionFlyout({
     }
   }, [rules]);
 
+  const showLoadingSkeleton = useMemo(
+    () => isLoading && isAlertDataLoading && !sharedListToAddTo && !rules?.length,
```

Comment:
so, loading skeleton would be shown only if all these conditions are true?
Wouldn't it mean, while rule is loading and there is still no full data, flyout will display part data only?
Like in this video


https://github.com/elastic/kibana/assets/92328789/a37aba11-2a43-4407-aae8-cab3a3545323

Shouldn't skeleton be displayed when all dependent data is loading, so the form will be rendered once we have all data?

Reply:
Yes this is true, I am planning to fix it in another PR and refactor the loading part, but this would require introducing new properties and expanding the scope of this PR, I just wanted to fix the loading that depends on the external factor because it causes sometimes losing data

Reply:
can we add `OR` condition for loaders?
loading skeleton should appear if any data page depends on still loading.
In this case it appears only if all data is still loading at the same moment of time.
Which could lead to situations, like in recording - displaying partly loaded page

Also, I think loading status should not depends on number of loaded rules `rules?.length` - if request failed and not rules fetched, error should be shown, instead of loader

> the external factor because it causes sometimes losing data
what do you mean by losing data?

Reply:
I handled the populating case only for now since it is applicable in one place https://github.com/elastic/kibana/pull/166611/commits/aca01f69e428068c744dfc9b0fbdd8792288fd15#diff-10f41a69a528b4fee8e9c3a0308519b55d34e6a8d7bf5a245113a59477883e5aR132, this will removing the loader before the population ends

Reply:
I can see loader is not flickering

But, looks like there 2 types of loaders visible now, which creates so called cascading loading of flyout.
In video skeleton can be seen first, that round spinner in place of fields components for a brief moment.
Which, something I believe probably not related to this PR and we can address later, as it happens only when the first alert is opened.

Another issue though, looks more severe and can't be reproduced on main
When I switch tabs, loading skeleton appears again and wipes out any changes to the form


https://github.com/elastic/kibana/assets/92328789/6d5eda9e-8a0b-43b2-8fa4-9eaaca9d8e72

Reply:
in main though, every time I switch tabs, there is rule request fired. It does not though, lead to any form wipes out

This is video from main

https://github.com/elastic/kibana/assets/92328789/5c0f1f3d-3040-417b-8f17-c381b7884024

Reply:
Since fixing part of the loading will cause unexpected issues, @vitaliidm and I decided to tackle the loading in another PR since the main bug was not related to the loader and the fix has to be merged in 8.10

Reply:
Will create an enhancement ticket for reducing the loader flicker behaviour

Reply:
Hey @WafaaNasr @vitaliidm, just FYI that I filed [this bug](https://github.com/elastic/kibana/issues/170081) related to the behavior described above. It was reported by a customer today.